### PR TITLE
[CP-943] replacing datasources

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-943]

**Description**

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-943]: https://appnroll.atlassian.net/browse/CP-943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/messages/reducers/messages.reducer.ts [hunk 1]
```diff
@@ -269,5 +271,31 @@ export const messagesReducer = createReducer<MessagesState>(
...
+                const messageIds: string[] = prev[message.threadId] ?? []
+                prev[message.threadId] = messageIds.find(
+                  (id) => id === message.id
+                )
+                  ? messageIds
+                  : [...messageIds, message.id]
+
+                return prev
+              }, {}),
+            threadsTotalCount: Object.keys(action.payload.messages).length - 1,
```
Comment (dkarski) [inline packages/app/src/messages/reducers/messages.reducer.ts:296]:
```suggestion
            threadsTotalCount: Object.keys(action.payload.threads).length,
            threadsState: ResultState.Loaded,
```

Comment (igorbezsmertnyi) [reply packages/app/src/messages/reducers/messages.reducer.ts:296]: @dkarski It's how it looks like without `-1`
Screen Shot 2022-01-21 at 8 26 44 AM

File: packages/app/src/messages/reducers/messages.reducer.ts [hunk 2]
```diff
@@ -269,5 +271,32 @@ export const messagesReducer = createReducer<MessagesState>(
           messageIdsInThreadMap: {},
         }
       })
+
+      .addCase(
```
Comment (dkarski) [inline packages/app/src/messages/reducers/messages.reducer.ts:275]: please modify MessagesEvent.SetMessages, to mutate state instead of clear

**Review by dkarski — COMMENTED** [review]
please uncomment interval in `packages/app/src/messages/components/messages/messages.component.tsx`

**Review by dkarski — APPROVED** [review]

**Pull request merged**